packages/react/src/components/Comments/ThreadsSidebar.tsx

```diff
@@ -0,0 +1,188 @@
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useUIPluginState } from "../../hooks/useUIPluginState.js";
+import { Thread } from "./Thread.js";
+import { useThreads } from "./useThreads.js";
+import { ThreadData } from "@blocknote/core/comments";
+import { BlockNoteEditor } from "@blocknote/core";
+import { useEffect, useMemo } from "react";
+import { TextSelection } from "@tiptap/pm/state";
+
+function sortThreads(
+  editor: BlockNoteEditor<any, any, any>,
+  threads: ThreadData[],
+  sort?: "position" | "newest" | "oldest"
+) {
+  if (sort === "position") {
+    return threads.sort((a, b) => {
+      const threadA =
+        editor.comments?.plugin
+          .getState(editor.prosemirrorState)
+          .threadPositions.get(a.id)?.from || Number.MAX_VALUE;
+      const threadB =
+        editor.comments?.plugin
+          .getState(editor.prosemirrorState)
+          .threadPositions.get(b.id)?.from || Number.MAX_VALUE;
+
+      return threadA - threadB;
+    });
+  }
+
+  if (sort === "newest") {
+    return threads.sort((a, b) => {
+      return b.createdAt.getTime() - a.createdAt.getTime();
+    });
+  }
+
+  if (sort === "oldest") {
+    return threads.sort((a, b) => {
+      return a.createdAt.getTime() - b.createdAt.getTime();
+    });
+  }
+
+  return threads;
+}
+
+/**
+ * The ThreadsSidebar component can be used to display a list of comments in a sidebar.
+ *
+ * This component is similar to Google Docs "Show All Comments" sidebar (cmd+option+shift+A)
+ */
+export function ThreadsSidebar(props: {
+  filter?: "open" | "resolved";
+  // TODO: Should be implemented for both floating and sidebar views
+  // maxCommentsPerThread?: number;
+  sort?: "position" | "newest" | "oldest";
+}) {
+  const editor = useBlockNoteEditor();
+
+  if (!editor.comments) {
+    throw new Error("Comments plugin not found");
+  }
+
+  const state = useUIPluginState(
+    editor.comments.onUpdate.bind(editor.comments)
+  );
+  const selectedThreadId = state?.selectedThreadId;
+
+  const threads = useThreads(editor);
+
+  const threadPositions = useMemo<
+    | Map<
+        string,
+        {
+          from: number;
+          to: number;
+        }
+      >
+    | undefined
+  >(
+    () =>
+      editor.comments?.plugin.getState(editor.prosemirrorState)
+        ?.threadPositions,
+    [editor.comments?.plugin, editor.prosemirrorState]
```

Comment:
I don't think this would update after a state update, right? (because there won't be a rerender when the state updates. We could also follow the existing pattern by accessing this via  `useUIPluginState`?